f8a_worker/workers/recommender.py

```diff
@@ -639,6 +642,78 @@ def _get_stack_values(self, similar_stacks_list):
             similarity_list.append(s_stack)
         return similarity_list
 
+def invoke_license_analysis_service(user_stack_packages, alternate_packages, companion_packages):
+    license_url = LICENSE_SCORING_URL_REST + "/api/v1/stack_license"
+
+    payload = {
+        "packages": user_stack_packages,
+        "alternate_packages": alternate_packages,
+        "companion_packages": companion_packages
+    }
+
+    resp = {}
+    try:
+        license_req = get_session_retry().post(license_url, data=json.dumps(payload))
+        resp = license_req.json()
+    except:  # TODO customized exception
+        msg = traceback.format_exc()
+        _logger.error("Unexpected error happened while invoking license analysis!\n{}".format(msg))
+
+    return resp
+
+def apply_license_filter(user_stack_components, epv_list_alt, epv_list_com):
+    license_score_list_alt = []
+    for epv in epv_list_alt:
+        license_scoring_input = {
+            'package': epv.get('pkg', {}).get('name', [''])[0],
+            'version': epv.get('ver', {}).get('version', [''])[0],
+            'licenses': epv.get('ver', {}).get('version', {}).get('licenses', [])
+        }
+        license_score_list_alt.append(license_scoring_input)
+
+    license_score_list_com = []
+    for epv in epv_list_com:
+        license_scoring_input = {
+            'package': epv.get('pkg', {}).get('name', [''])[0],
+            'version': epv.get('ver', {}).get('version', [''])[0],
+            'licenses': epv.get('ver', {}).get('version', {}).get('licenses', [])
+        }
+        license_score_list_com.append(license_scoring_input)
+
+    # Call license scoring to find license filters
+    la_output = invoke_license_analysis_service(user_stack_components,
+                                                license_score_list_alt,
+                                                license_score_list_com)
+
+    conflict_packages_alt = conflict_packages_com = []
+    if la_output['status'] == 'Successful' and la_output['license_filter'] is not None:
+        license_filter = la_output.get('license_filter', {})
+        conflict_packages_alt = license_filter.get('alternate_packages', {}).get('conflict_packages', [])
+        conflict_packages_com = license_filter.get('companion_packages', {}).get('conflict_packages', [])
+
+    list_pkg_names_alt = []
+    for epv in epv_list_alt:
+        name = epv.get('pkg', {}).get('name', [''])[0]
+        if name in conflict_packages_alt:
+            list_pkg_names_alt.append(name)
+            epv_list_alt.remove(epv)
+
+    list_pkg_names_com = []
+    for epv in epv_list_com:
+        name = epv.get('pkg', {}).get('name', [''])[0]
+        if name in conflict_packages_com:
+            list_pkg_names_com.append(name)
+            epv_list_com.remove(epv)
```

Comment:
The same as above.

Reply:
Done. I now use `for epv in epv_list_com[:]:`